src/lib/edge-functions/proxy.mjs

```diff
@@ -88,26 +88,30 @@ export const initializeProxy = async ({
     port: isolatePort,
     projectDir,
   })
-  const hasEdgeFunctions = userFunctionsPath !== undefined || internalFunctions.length !== 0
+
+  if (!userFunctionsPath && !internalFunctionsPath) {
+    return
+  }
 
   return async (req) => {
+    const registry = await server
```

Comment:
You're still initialising the server and therefore downloading the Deno CLI, regardless of whether or not the site uses edge functions, which I thought you were going to change. Or am I missing something?

Reply:
@eduardoboucas I'm doing an empty return on line 92 if there is no userfunctionspath or internalfunctionspath. I thought this would ensure that line 97 would not be run if we didn't need a server?

Reply:
How I see this I think this is a little bit of a egg-hen problem, without the server we do not know at this point if there a functions. Unless we extract the logic to find edge-functions out from the server/registry.